src/analysis/souffle/tests/test_taint_with_auth_logic.dl

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+#include "../taint_with_auth_logic.dl"
+#include "../flat_graph_ir.dl"
+#include "../edges_from_graph.dl"
+
+// This is a datalog file that is meant to test taint_with_auth_logic.dl. It 
+// should eventually be replaced by or otherwise used as a part of an
+// automated test.
+
+//-----------------------------------------------------------------------------
+// Program Manifests
+//-----------------------------------------------------------------------------
+
+// Module 1
+//      reads: 
+//          some_struct{ a, b }
+//          other_input
+//      writes:
+//          mod1out
+bindFlatParticle("module1", $FlatParticle(
+    /* reads */
+    $Cons("some_struct.a", $Cons("some_struct.b",
+        $Cons("other_input", $Nil()))),
+    /* writes */
+    $Cons("mod1out", $Nil())
+)).
+
+// the label of "other_input" is the empty set (which makes it public).
+// so it has no label elements
+
+// Module 2
+//     reads:
+//         mod1out
+//         input4
+//     writes:
+//         mod2out 
+bindFlatParticle("module2", $FlatParticle(
+    /* reads */
+    $Cons("mod1out", $Cons("input4", $Nil())),
+    /* writes */
+    $Cons("mod2out", $Nil())
+)).
+
+//-----------------------------------------------------------------------------
+// Policy
+//-----------------------------------------------------------------------------
+
+// Assign owners to tags
+ownsTag("Alice", "raw_video").
+ownsTag("Alice", "image_selection").
+ownsTag("Alice", "product_id").
+ownsTag("ServiceProvider", "model_weights").
+
+// Assign tags to initial inputs
+claimHasTag("some_struct.a", "raw_video").
```

Comment:
All of these tags are secrecy tags. Just adding these claims is not correct if these are integrity tags, right?

Reply:
Based on our discussion about timestamps, I believe we agreed that we will need integrity labels even with authorization logic. For example, we could make downgrades conditional:
```
downgrades("x", "blah") :- hasIntegrity("x", "blahblah"). 
```

Reply:
OK we can do integrity tags. I think that's substantial enough that adding integrity tags should go in another PR though.